ladybug/datacollection.py

```diff
@@ -406,7 +406,7 @@ def filter_by_hoys(self, hoys):
            filteredDBT = DBT.filter_by_hoys(hoys)
         """
         _moys = tuple(int(hour * 60) for hour in hoys)
-        return self.filter_by_moys(_moys)
+        return DataCollection(self.filter_by_moys(_moys))
```

Comment:
Maybe it is a better idea if `filter_by_moys` returns a DataCollection in the first place.